### PR TITLE
fix: prevent git ignore dev-bundle files

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

The `build` rule prevents part of the dev-bundle to being added to the git repository. This change adds an exclusion to the `.gitignore` file to prevent part of the bundle to being silently ignored

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 build
+!**/dev-bundle/webapp/VAADIN/build
```
Comment (mshabarov) [inline .gitignore:2]: Why don't ignore all the files inside dev bundle, like this `!**/dev-bundle`? Assets and stats.json have to be committed to VCS as well.

**Comment by mshabarov** [comment]
~This can be picked to other starters in `v24` branches.~
Needed only for gradle projects.

File: .gitignore (hunk 1 above)
Comment (mcollovati) [reply .gitignore:2]: The problem here is that in gradle project we have a rule to ignore `build`
If I'm not mistaken, the `!**/dev-bundle` rule would not prevent the `/dev-bundle/webapp/VAADIN/build` from being ignored.
The rule needs to end with the `build` segment

Comment (mcollovati) [reply .gitignore:2]: You can check it by creating a temp file in `dev-bundle/webapp/VAADIN/build/`, e.g. `touch src/main/dev-bundle/webapp/VAADIN/build/test.me` and then run `git check-ignore -v src/main/dev-bundle/webapp/VAADIN/build/test.me`

With the current rule, the command produces no output, but if we change it to `!**/dev-bundle`, it prints
```
.gitignore:1:build	src/main/dev-bundle/webapp/VAADIN/build/test.me
```

where `.gitignore:1:build` is a reference to the blocking rule `build`, defined at row 1 of `.gitignore` file

**Review by mshabarov — APPROVED** [review]

**Pull request merged**